src/EnergyPlus/ChilledCeilingPanelSimple.cc

```diff
@@ -0,0 +1,1460 @@
+					ThisSurfIntensity = ( CoolingPanelSource( CoolingPanelNum ) * CoolingPanel( CoolingPanelNum ).FracDistribToSurf( RadSurfNum ) / Surface( SurfNum ).Area );
+					QCoolingPanelSurf( SurfNum ) += ThisSurfIntensity;
+					// CR 8074, trap for excessive intensity (throws off surface balance )
+					if ( ThisSurfIntensity > MaxRadHeatFlux ) {
+						ShowSevereError( "DistributeCoolingPanelRadGains:  excessive thermal radiation heat flux intensity detected" );
+						ShowContinueError( "Surface = " + Surface( SurfNum ).Name );
+						ShowContinueError( "Surface area = " + RoundSigDigits( Surface( SurfNum ).Area, 3 ) + " [m2]" );
+						ShowContinueError( "Occurs in " + cCMO_CoolingPanel_Simple + " = " + CoolingPanel( CoolingPanelNum ).EquipID );
+						ShowContinueError( "Radiation intensity = " + RoundSigDigits( ThisSurfIntensity, 2 ) + " [W/m2]" );
+						ShowContinueError( "Assign a larger surface area or more surfaces in " + cCMO_CoolingPanel_Simple );
+						ShowFatalError( "DistributeCoolingPanelRadGains:  excessive thermal radiation heat flux intensity detected" );
+					}
+				} else {
+					ShowSevereError( "DistributeCoolingPanelRadGains:  surface not large enough to receive thermal radiation heat flux" );
+					ShowContinueError( "Surface = " + Surface( SurfNum ).Name );
+					ShowContinueError( "Surface area = " + RoundSigDigits( Surface( SurfNum ).Area, 3 ) + " [m2]" );
+					ShowContinueError( "Occurs in " + cCMO_CoolingPanel_Simple + " = " + CoolingPanel( CoolingPanelNum ).EquipID );
+					ShowContinueError( "Assign a larger surface area or more surfaces in " + cCMO_CoolingPanel_Simple );
+					ShowFatalError( "DistributeCoolingPanelRadGains:  surface not large enough to receive thermal radiation heat flux" );
+
+				}
+			}
+
+		}
+
+	}
+
+	void
+	ReportCoolingPanel( int const CoolingPanelNum )
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Daeho Kang
+		//       DATE WRITTEN   Aug 2007
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE: This subroutine
+
+		// METHODOLOGY EMPLOYED:
+		// na
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+		using DataLoopNode::Node;
+		using DataSurfaces::Surface;
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS
+		// na
+
+		// DERIVED TYPE DEFINITIONS
+		// na
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+
+		CoolingPanel( CoolingPanelNum ).TotEnergy = CoolingPanel( CoolingPanelNum ).TotPower * TimeStepSys * SecInHour;
+		CoolingPanel( CoolingPanelNum ).Energy = CoolingPanel( CoolingPanelNum ).Power * TimeStepSys * SecInHour;
+		CoolingPanel( CoolingPanelNum ).ConvEnergy = CoolingPanel( CoolingPanelNum ).ConvPower * TimeStepSys * SecInHour;
+		CoolingPanel( CoolingPanelNum ).RadEnergy = CoolingPanel( CoolingPanelNum ).RadPower * TimeStepSys * SecInHour;
+
+	}
+
+	Real64
+	SumHATsurf( int const ZoneNum ) // Zone number
```

Comment:
@Myoldmopar Well, that is a good idea and would take a bit to implement because it is not just an issue for the new chilled ceiling but also every other system that has followed in the footsteps of the radiant model.  That's actually a lot of stuff including various baseboard models, ventilated slab, etc.  So, are you wanting this done for all of these now or just for the chilled ceiling or this handled as some sort of defect/improvement?  I definitely agree now that you have brought this up that it would be better from the standpoint of duplicated code to implement this as a publicly available function as I believe everything outside the surface heat balance routines follows the exact same method.